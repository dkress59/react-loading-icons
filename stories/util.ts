import { ArgTypes } from '@storybook/react'
import { SVGProps } from 'react'

export type IconProps = JSX.IntrinsicAttributes & SVGProps<SVGSVGElement>
export const IconFill: IconProps = {
	fill: '#06bcee',
	fillOpacity: 1,
	height: '3em',
	speed: 1,
	stroke: 'transparent',
	strokeOpacity: 1,
	width: 'auto',
}
export const IconStroke: IconProps = {
	fill: 'transparent',
	fillOpacity: 1,
	height: '3em',
	speed: 1,
	stroke: '#06bcee',
	strokeOpacity: 1,
	strokeWidth: 2,
	width: 'auto',
}
export const IconStrokeFill: IconProps = {
	fill: '#06bcee',
	fillOpacity: 1,
	height: '3em',
	speed: 1,
	stroke: '#06bcee',
	strokeOpacity: 1,
	strokeWidth: 2,
	width: 'auto',
}

export const IconFillArgs: ArgTypes = {
	height: {
		name: 'Height',
		control: 'text',
		defaultValue: '3em',
	},
	width: {
		name: 'Width',
		control: 'text',
		defaultValue: 'auto',
	},
	fill: {
		name: 'Fill',
		control: 'color',
		defaultValue: '#06bcee',
	},
	stroke: {
		name: 'Stroke',
		control: 'color',
		defaultValue: 'transparent',
	},
	speed: {
		name: 'Speed',
		control: 'number',
		defaultValue: 1,
	},
}
export const IconStrokeArgs: ArgTypes = {
	height: {
		name: 'Height',
		control: 'text',
		defaultValue: '3em',
	},
	width: {
		name: 'Width',
		control: 'text',
		defaultValue: 'auto',
	},
	fill: {
		name: 'Fill',
		control: 'color',
		defaultValue: 'transparent',
	},
	stroke: {
		name: 'Stroke',
		control: 'color',
		defaultValue: '#06bcee',
	},
	speed: {
		name: 'Speed',
		control: 'number',
		defaultValue: 1,
	},
}
export const IconStrokeFillArgs: ArgTypes = {
	height: {
		name: 'Height',
		control: 'text',
		defaultValue: '3em',
	},
	width: {
		name: 'Width',
		control: 'text',
		defaultValue: 'auto',
	},
	fill: {
		name: 'Fill',
		control: 'color',
		defaultValue: '#06bcee',
	},
	stroke: {
		name: 'Stroke',
		control: 'color',
		defaultValue: '#06bcee',
	},
	speed: {
		name: 'Speed',
		control: 'number',
		defaultValue: 1,
	},
}
