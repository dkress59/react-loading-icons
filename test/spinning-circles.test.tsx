import { render, screen } from '@testing-library/react'
import React from 'react'

import SpinningCircles from '../src/components/spinning-circles'
import { testProps } from './util'

describe('<SpinningCircles /> component', () => {
	it('renders and matches snapshot', () => {
		render(<SpinningCircles data-testid="icon" />)
		const svg = screen.getByTestId('icon')
		expect(svg).not.toBeNull()
		expect(svg).toMatchSnapshot()
		expect(svg.getAttributeNames()).toContain('viewBox')
		expect(svg.getAttribute('class')).toBe('icon-loading')
	})
	it('inherits props', () => {
		render(<SpinningCircles {...testProps} />)
		const svg = screen.getByTestId('icon')
		expect(svg.getAttribute('height')).toBe('2em')
		expect(svg.getAttribute('width')).toBe('auto')
		expect(svg.getAttribute('class')).toBe('icon-loading test')
	})
})
