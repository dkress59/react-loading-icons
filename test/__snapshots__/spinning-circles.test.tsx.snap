// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpinningCircles /> component renders and matches snapshot 1`] = `
<svg
  class="icon-loading"
  data-testid="icon"
  height="58"
  viewBox="0 0 58 58"
  width="58"
>
  <g
    fill="#fff"
    fill-rule="evenodd"
    transform="translate(2 1)"
  >
    <circle
      cx="42.601"
      cy="11.462"
      r="5"
    >
      <animate
        attributeName="fill-opacity"
        begin="0s"
        calcMode="linear"
        dur="1.3s"
        repeatCount="indefinite"
        values="1;0;0;0;0;0;0;0"
      />
    </circle>
    <circle
      cx="49.063"
      cy="27.063"
      fill-opacity="0"
      r="5"
    >
      <animate
        attributeName="fill-opacity"
        begin="0s"
        calcMode="linear"
        dur="1.3s"
        repeatCount="indefinite"
        values="0;1;0;0;0;0;0;0"
      />
    </circle>
    <circle
      cx="42.601"
      cy="42.663"
      fill-opacity="0"
      r="5"
    >
      <animate
        attributeName="fill-opacity"
        begin="0s"
        calcMode="linear"
        dur="1.3s"
        repeatCount="indefinite"
        values="0;0;1;0;0;0;0;0"
      />
    </circle>
    <circle
      cx="27"
      cy="49.125"
      fill-opacity="0"
      r="5"
    >
      <animate
        attributeName="fill-opacity"
        begin="0s"
        calcMode="linear"
        dur="1.3s"
        repeatCount="indefinite"
        values="0;0;0;1;0;0;0;0"
      />
    </circle>
    <circle
      cx="11.399"
      cy="42.663"
      fill-opacity="0"
      r="5"
    >
      <animate
        attributeName="fill-opacity"
        begin="0s"
        calcMode="linear"
        dur="1.3s"
        repeatCount="indefinite"
        values="0;0;0;0;1;0;0;0"
      />
    </circle>
    <circle
      cx="4.938"
      cy="27.063"
      fill-opacity="0"
      r="5"
    >
      <animate
        attributeName="fill-opacity"
        begin="0s"
        calcMode="linear"
        dur="1.3s"
        repeatCount="indefinite"
        values="0;0;0;0;0;1;0;0"
      />
    </circle>
    <circle
      cx="11.399"
      cy="11.462"
      fill-opacity="0"
      r="5"
    >
      <animate
        attributeName="fill-opacity"
        begin="0s"
        calcMode="linear"
        dur="1.3s"
        repeatCount="indefinite"
        values="0;0;0;0;0;0;1;0"
      />
    </circle>
    <circle
      cx="27"
      cy="5"
      fill-opacity="0"
      r="5"
    >
      <animate
        attributeName="fill-opacity"
        begin="0s"
        calcMode="linear"
        dur="1.3s"
        repeatCount="indefinite"
        values="0;0;0;0;0;0;0;1"
      />
    </circle>
  </g>
</svg>
`;
